typing/predef.ml

```diff
@@ -163,54 +236,86 @@ let mk_add_type add_type type_ident ?manifest
      type_uid = Uid.of_predef_id type_ident;
     }
   in
-  add_type type_ident decl env
-
-let build_initial_env add_type add_extension empty_env =
-  let add_type = mk_add_type add_type
-  and add_type1 type_ident
-      ~variance ~separability ?(kind=fun _ -> Type_abstract Definition) env =
+  let decl1
+      ~variance
+      ?(separability = Separability.Ind)
+      ?(kind = fun _ -> Type_abstract Definition)
+      type_ident
+    =
     let param = newgenvar () in
-    let decl =
-      {type_params = [param];
-       type_arity = 1;
-       type_kind = kind param;
-       type_loc = Location.none;
-       type_private = Asttypes.Public;
-       type_manifest = None;
-       type_variance = [variance];
-       type_separability = [separability];
-       type_is_newtype = false;
-       type_expansion_scope = lowest_level;
-       type_attributes = [];
-       type_immediate = Unknown;
-       type_unboxed_default = false;
-       type_uid = Uid.of_predef_id type_ident;
-      }
-    in
-    add_type type_ident decl env
-  and add_continuation type_ident env =
-    let tvar1 = newgenvar() in
-    let tvar2 = newgenvar() in
-    let arity = 2 in
-    let decl =
-      {type_params = [tvar1; tvar2];
-       type_arity = arity;
-       type_kind = Type_abstract Definition;
-       type_loc = Location.none;
-       type_private = Asttypes.Public;
-       type_manifest = None;
-       type_variance = [Variance.contravariant; Variance.covariant];
-       type_separability = Types.Separability.default_signature ~arity;
-       type_is_newtype = false;
-       type_expansion_scope = lowest_level;
-       type_attributes = [];
-       type_immediate = Unknown;
-       type_unboxed_default = false;
-       type_uid = Uid.of_predef_id type_ident;
-      }
-    in
-    add_type type_ident decl env
+    { (decl0 ~kind:(kind param) type_ident) with
+      type_params = [param];
+      type_arity = 1;
+      type_variance = [variance];
+      type_separability = [separability];
+    }
   in
+  let decl2
+      ~variance:(var1, var2)
+      ?separability:((sep1, sep2) = (Separability.Ind, Separability.Ind))
```

Comment:
This argument (and the same one in `decl1`) is never set to another value than the default one, maybe we should just remove it ?

Reply:
For now I propose to keep them around. The distinction I make is that the current API lets you specify all parameters you may need to change. If someone wants to add a type with a different separability (for example, an unsafe type of unboxed options, which should be `Sep` as it can return its argument type unboxed), right now it is clear how to do it. If I remove the parameters, then this person will have to choose between re-adding the parameter (consistently at all higher-arity makers? it's a chore) or override the field manually, which is a code smell as it breaks the smart-constructor abstraction.